bin/gwdetchar-omega

```diff
@@ -153,6 +154,51 @@ class OmegaChannelList(object):
 
 # -- Utilities ----------------------------------------------------------------
 
+def get_data(channels, time, duration, pad, frametype=None, source=None,
+             nproc=1, verbose=False):
+    """Retrieve data for multiple channels, centered at a given time
+
+    Parameters
+    ----------
+    channels : `list`
+        required data channels
+    time : `float`
+        GPS time of required data
+    duration : `float`
+        duration (in seconds) of required data
+    pad : `float`
+        amount of extra data to read in at the start and end for filtering
+    frametype : `str`, optional
+        name of frametype in which this channel is stored, by default will
+        search for all required frame types
+    source : `str`, `list`, optional
+        `str` path of a LAL-format cache file or single data file, will
+        supercede `frametype` if given, defaults to `None`
+    nproc : `int`, optional
+        number of parallel processes to use, uses serial process by default
+    verbose : `bool`, optional
+        print verbose output about NDS progress, default: False
+
+    See Also
+    --------
+    gwpy.timeseries.TimeSeriesDict.get
+        for the underlying method to read from frames or NDS
+    gwpy.timeseries.TimeSeriesDict.read
+        for the underlying method to read from a local file cache
+    """
+    # set GPS start and end time
+    start = time - duration/2. - pad
+    end = time + duration/2. + pad
+    # read from cache if requested
+    if source:
+        data = TimeSeriesDict.read(source, channels, start=start, end=end,
+                                   nproc=nproc)
+    # read from frames or NDS
+    else:
+        data = TimeSeriesDict.get(channels, start, end, frametype=frametype,
+                                  nproc=nproc, verbose=verbose)
+    return data
```

Comment:
@alurban, I would refactor this a little, mainly for cleanliness as:

```python
    # read from cache if requested
    if source:
        return TimeSeriesDict.read(source, channels, start=start, end=end,
                                   nproc=nproc)
    # read from frames or NDS
    return TimeSeriesDict.get(channels, start, end, frametype=frametype,
                              nproc=nproc, verbose=verbose)
```

Reply:
@duncanmmacleod, good suggestion! I made this change and double-checked it by updating the test page.